src/aatoolbox/datasources/iri/iri_forecast_seasonal_precipitation_tercile.py

```diff
@@ -0,0 +1,184 @@
+        be included in the data.
+    """
+
+    def __init__(self, iso3, geo_bounding_box: GeoBoundingBox):
+        super().__init__(
+            iso3=iso3, module_base_dir=_MODULE_BASENAME, is_public=False
+        )
+        self._iso3 = iso3
+        # round coordinates to correspond with the grid IRI publishes
+        # its data on, which is 1 degree resolution
+        # non-rounded coordinates can be given to the URL which then
+        # automatically rounds them, but for file saving we prefer to do
+        # this ourselves
+        geo_bounding_box.round_coords(round_val=1, offset_val=0)
+        self._geobb = geo_bounding_box
+
+    def download(
+        self,
+        iri_auth: str,
+        # question: I chose to make dominant a function arg
+        # instead of class attr. Does this make sense?
+        # Does it even make sense to have dominant as an arg
+        # or should we have a separate class for it?
+        dominant: bool = False,
+    ):
+        """
+        Download the IRI seasonal tercile forecast as NetCDF file.
+
+        Parameters
+        ----------
+        iri_auth: str
+            iri key for authentication. An account is
+            needed to get this key config. For an account this key might
+            be changed over time, so might need to update it regularly
+        dominant: bool
+            Two datasets with the tercile forecast exists. One reports
+            the probability per tercile, which is downloaded with
+            dominant=False. The other only reports the probability
+            of the dominant tercile, which is downloaded with
+            dominant=True. In this data low negative probabilities indicate the
+            below average tercile is dominant and high positive that the above
+            average tercile is dominant.
+
+        Examples
+        --------
+        >>> from aatoolbox.pipeline import Pipeline
+        >>> from aatoolbox.utils.geoboundingbox import GeoBoundingBox
+        >>> (from aatoolbox.datasources.iri.
+        ... iri_forecast_seasonal_precipitation_tercile import IriForecast)
+        >>> iso3="bfa"
+        >>> pipeline_iso = Pipeline(iso3)
+        >>> codab_admin1 = pipeline_iso.load_codab(admin_level=1)
+        >>> geo_bounding_box = GeoBoundingBox.from_shape(codab_admin1)
+        >>> iri=IriForecast(iso3,geo_bounding_box)
+        >>> iri.download(os.getenv("IRI_AUTH"))
+        """
+        # question: I chose to always remove the file and redownload
+        # for all dates instead of allowing a selection of dates.
+        # This because the file is <1MB and only takes about
+        # a minute to download. Do you think this makes sense?
+
+        output_filepath = self._get_raw_path(dominant=dominant)
+        # strange things happen when just overwriting the file, so delete it
+        # first if it already exists
+        output_filepath.unlink(missing_ok=True)
+        output_filepath.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
Is there a way you can remove the file without needing to recreate the directories?

Reply:
Ah I think the order is confusing. The mkdir is in case the file and the parent dir don't exist in the first place. The unlink only removes the file. I changed the order of the two lines. Does that make sense?

Reply:
Ah sorry yes that's clear now